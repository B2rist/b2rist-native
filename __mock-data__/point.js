export const points = [
  {
    user: 'test@gmail.com',
    contentType: 'sound-collage',
    name: 'Elephants',
    description:
      'Elephants are the heaviest land mammals in the world. The male African elephant can reach a weight of up to 6 tonnes. The Asian elephant, smaller than it’s African relative, has been domesticated for thousands of years, often being used as pack-animals. Katka and Wiki live in Oliwa Zoo and are rather elderly representatives of both species. They live together in the pavilion, which was prepared for the elephants in the early 50s.  The African elephant Katka was born around 1970 and came to us from France. She presented her excellent memory by recognizing her Czech keeper after not seeing him for 16 years!  The Asian elephant Wiki was born in 1964 and came to Oliwa zoo under very tragic circumstances after animal rights protection organisation retrieved her from a closed circus. She found her new home here with us in Gdansk. After the traumatic experience of working in a circus, Wiki needed two years in order to trust people again',
    radius: 20,
    isRoute: false,
    id: 'm3Ss7dha9TRCT8na9irS',
    location: {
      _latitude: 54.41221902644764,
      _longitude: 18.533400003253522,
    },
    cover: '1QxEYG2Z_yUrltO2S02aZC3q39WtlO1bj',
    geohash: 'u3tjt9rw5s',
    language: 'en',
    audioContent: '1SPT9SwLvr89EFdxydfZ7yQ4d_jEE9yvd',
    images: [
      {
        time: 0,
        image: '1QxEYG2Z_yUrltO2S02aZC3q39WtlO1bj',
      },
    ],
  },
  {
    user: 'test@gmail.com',
    contentType: 'sound-collage',
    name: 'Lions',
    description:
      'Lives in southwest Africa and is the biggest predator of the continent. It can be found in various environments: forested, semi-desert and even mountainous areas at the altitude of 4000m. Unlike other wild cats, lions live in pride consisting of up to 30 individuals. Dominant male is responsible for protection of offspring and his territory while females prey for food and care for their cubs. Most of their time lions spend resting and sleeping. Although lions are able to swim they avoid water.',
    radius: 20,
    isRoute: false,
    id: '0wcA2uvk3yVHBrqpRKwB',
    location: {
      _latitude: 54.410370053735996,
      _longitude: 18.539548069238663,
    },
    cover: '1N3cjoN3DBovGiySj6IwHCclBdyFH6Lks',
    geohash: 'u3tjtch5cq',
    language: 'en',
    audioContent: '1SPT9SwLvr89EFdxydfZ7yQ4d_jEE9yvd',
    images: [
      {
        time: 0,
        image: '1N3cjoN3DBovGiySj6IwHCclBdyFH6Lks',
      },
    ],
  },
  {
    user: 'test@gmail.com',
    contentType: 'sound-collage',
    name: 'Green gate',
    description:
      "With the Golden Gate and the Highland Gate, the Green Gate spans the Long Market and Long Street, together comprising the Royal Route. The Green Gate was clearly inspired by the Antwerp City Hall.[1] It was built 1568-71 as the formal residence of Poland's monarchs.[2] It is a masterpiece by Regnier (or Reiner van Amsterdam), an Amsterdam architect,[3] and reflects Flemish architectural influence in Gdańsk. Hans Kramer from Dresden was responsible for the construction plans.  On 11–20 February 1646 the future Queen of Poland, Marie Louise Gonzaga, was entertained here. In the late 18th century the Nature Society was housed here, but soon moved to the Naturalists' House (Research Society House).  Today the Green Gate houses the National Museum in Gdańsk. Exhibitions, meetings, conferences and shows are held here. The Gdańsk office of former Polish President Lech Wałęsa is located in one of the rooms.",
    radius: 20,
    isRoute: false,
    id: 'L5Bq5UzXl1EkAKj9E4xf',
    location: {
      _latitude: 54.34817976567255,
      _longitude: 18.65583504157628,
    },
    cover: '1Dr1ml2-GZcA5zyf68VmiFudFAoyCu9eb',
    geohash: 'u3tm0tbc07',
    language: 'en',
    audioContent: '1SPT9SwLvr89EFdxydfZ7yQ4d_jEE9yvd',
    images: [
      {
        time: 0,
        image: '1Dr1ml2-GZcA5zyf68VmiFudFAoyCu9eb',
      },
      {
        time: 10,
        image: '1JXNtJrUOfYppjEpzIBgQ7EWTs31Bs-Dy',
      },
    ],
  },
  {
    user: 'test@gmail.com',
    contentType: 'sound-collage',
    name: 'Stocznia Gdańska',
    description:
      'Brama nr 2 Stoczni Gdańskiej – jedna z bram prowadzących do Stoczni Gdańskiej. Ze względu na bliskość budynków dyrekcji stoczni oraz dogodne połączenie ze Śródmieściem Gdańska i dworcem kolejowym Gdańsk Główny powszechnie uznawana za główne wejście na teren zakładu.  W 1999 brama została wpisana do rejestru zabytków województwa pomorskiego pod numerem A-1206, a w 2014 została wyróżniona Znakiem Dziedzictwa Europejskiego w ramach historycznego zespołu obiektów Stoczni Gdańskiej',
    radius: 20,
    isRoute: false,
    id: 'C8U9O6WC9YQIsKVgpFzm',
    location: {
      _latitude: 54.360587751617885,
      _longitude: 18.649858281381743,
    },
    cover: '15MA3J9bHClcGGhoZBScMVVi3bxoS5_U3',
    geohash: 'u3tm22h3mh',
    language: 'en',
    audioContent: '1SPT9SwLvr89EFdxydfZ7yQ4d_jEE9yvd',
    images: [
      {
        time: 0,
        image: '15MA3J9bHClcGGhoZBScMVVi3bxoS5_U3',
      },
      {
        time: 10,
        image: '1qwvpjIsdsC-Nw1zP2o9lPi_6ZiGgBHYI',
      },
    ],
  },
];
