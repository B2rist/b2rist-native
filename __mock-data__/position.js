export const position = {
  speed: 0,
  heading: 0,
  altitude: 0,
  accuracy: 5.258999824523926,
  longitude: 18.65263,
  altitudeAccuracy: 0.5,
  latitude: 54.3486967,
};
