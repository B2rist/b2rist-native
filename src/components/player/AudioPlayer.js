import React, {useCallback, useEffect, useMemo, useState} from 'react';
import * as PropTypes from 'prop-types';
import {ActivityIndicator, IconButton, Text} from 'react-native-paper';
import Sound from 'react-native-sound';
import {
  useWindowDimensions,
  StyleSheet,
  ImageBackground,
  Platform,
  Slider,
  View,
  TouchableOpacity,
} from 'react-native';
import {audioLink, imageLink} from 'utils/googleLinks';

const getAudioTimeString = seconds => {
  const h = parseInt(seconds / (60 * 60), 10);
  const m = parseInt((seconds % (60 * 60)) / 60, 10);
  const s = parseInt(seconds % 60, 10);

  return (
    (h < 10 ? '0' + h : h) +
    ':' +
    (m < 10 ? '0' + m : m) +
    ':' +
    (s < 10 ? '0' + s : s)
  );
};

const AudioPlayer = ({
  point,
  close,
  showPrevious,
  showNext,
  isLast,
  disablePointControls,
  isFirst,
}) => {
  const {height, width} = useWindowDimensions();
  const timeMap = useMemo(() => {
    const images = [...point.images];
    images.sort((a, b) => {
      if (a.time < b.time) {
        return -1;
      }
      if (a.time > b.time) {
        return 1;
      }
      return 0;
    });
    return images;
  }, [point]);
  const [currentImage, setCurrentImage] = useState(timeMap[0]);
  const [sound, setSound] = useState(null);
  const [soundLoaded, setSoundLoaded] = useState(false);
  const [playState, setPlayState] = useState('paused');
  const [playSeconds, setPlaySeconds] = useState(0);
  const [sliderEditing, setSliderEditing] = useState(false);
  const [duration, setDuration] = useState(0);
  const [loop, setLoop] = useState(0);
  const styles = useMemo(() => createStyles({width, height}), [width, height]);

  const releaseSound = useCallback(() => {
    sound?.release();
  }, [sound]);

  const playComplete = useCallback(() => {
    setPlaySeconds(0);
    setPlayState('paused');
    if (sound) {
      sound.setCurrentTime(0);
    }
  }, [sound]);

  const onPlay = useCallback(() => {
    if (sound) {
      sound.play(playComplete);
      setPlayState('playing');
    } else {
      const loadSound = new Sound(
        audioLink(point.audioContent),
        null,
        error => {
          setSoundLoaded(true);
          if (error) {
            setPlayState('paused');
          } else {
            setSound(loadSound);
            setPlayState('playing');
            setDuration(loadSound.getDuration());
            loadSound.play(playComplete);
          }
        },
      );
    }
  }, [playComplete, point.audioContent, sound]);

  const setImageByTime = useCallback(
    time => {
      let imageKey = timeMap[0];
      for (const timeMapKey of timeMap) {
        if (time > timeMapKey.time) {
          imageKey = timeMapKey;
        } else {
          break;
        }
      }
      if (imageKey !== currentImage) {
        setCurrentImage(imageKey);
      }
    },
    [currentImage, timeMap],
  );

  useEffect(() => {
    onPlay();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [point]);

  useEffect(() => {
    return () => releaseSound();
  }, [releaseSound, sound]);

  useEffect(() => {
    const interval = setInterval(() => setLoop(Math.random()), 200);
    return () => clearInterval(interval);
  }, []);

  useEffect(() => {
    if (
      sound &&
      sound.isLoaded() &&
      playState === 'playing' &&
      !sliderEditing
    ) {
      sound.getCurrentTime(seconds => {
        setPlaySeconds(seconds);
        setImageByTime(seconds);
      });
    }
  }, [loop, playState, setImageByTime, sliderEditing, sound]);

  const currentTimeString = useMemo(
    () => getAudioTimeString(playSeconds),
    [playSeconds],
  );
  const durationString = useMemo(
    () => getAudioTimeString(duration),
    [duration],
  );

  const onClose = useCallback(() => {
    releaseSound();
    setSound(null);
    close();
  }, [close, releaseSound]);

  const onPause = useCallback(() => {
    sound?.pause();
    setPlayState('paused');
  }, [sound]);

  const onSliderEditStart = useCallback(() => {
    setSliderEditing(true);
  }, []);

  const onSliderEditEnd = useCallback(() => {
    setSliderEditing(false);
  }, []);

  const onSliderEditing = useCallback(
    value => {
      if (sound) {
        sound.setCurrentTime(value);
        setPlaySeconds(value);
      }
    },
    [sound],
  );

  const clickPause = useCallback(
    () => sound?.isLoaded() && (playState === 'paused' ? onPlay() : onPause()),
    [onPause, onPlay, playState, sound],
  );

  const onShowNext = useCallback(() => {
    sound?.release();
    setSound(null);
    showNext();
  }, [showNext, sound]);

  const onShowPrevious = useCallback(() => {
    sound?.release();
    setSound(null);
    showPrevious();
  }, [showPrevious, sound]);

  return (
    <TouchableOpacity
      activeOpacity={1}
      style={styles.container}
      onPress={clickPause}>
      <ImageBackground
        source={{uri: imageLink(currentImage.image)}}
        resizeMethod="resize"
        style={styles.image}>
        <View style={styles.closeButton}>
          <IconButton
            icon="close"
            size={50}
            onPress={onClose}
            disabled={!soundLoaded}
          />
        </View>
        {!disablePointControls && soundLoaded && playState === 'paused' && (
          <>
            <View style={styles.previousButton}>
              <IconButton
                icon="skip-previous"
                size={50}
                onPress={onShowPrevious}
                disabled={isFirst || !soundLoaded}
              />
            </View>
            <View style={styles.nextButton}>
              <IconButton
                icon="skip-next"
                size={50}
                onPress={onShowNext}
                disabled={isLast || !soundLoaded}
              />
            </View>
          </>
        )}
        <View style={styles.playButton}>
          {soundLoaded && playState === 'paused' && (
            <IconButton icon={'play'} size={50} onPress={clickPause} />
          )}
          {!soundLoaded && (
            <ActivityIndicator
              size={45}
              animating={true}
              style={styles.loadingSpinner}
            />
          )}
        </View>
        <View style={styles.controls}>
          <Text style={styles.time}>{currentTimeString}</Text>
          <Slider
            onTouchStart={onSliderEditStart}
            onTouchEnd={onSliderEditEnd}
            onValueChange={onSliderEditing}
            onSlidingComplete={onSliderEditing}
            value={playSeconds}
            maximumValue={duration}
            maximumTrackTintColor="gray"
            minimumTrackTintColor="white"
            thumbTintColor="white"
            style={styles.slider}
          />
          <Text style={styles.duration}>{durationString}</Text>
        </View>
      </ImageBackground>
    </TouchableOpacity>
  );
};

const createStyles = ({width, height}) =>
  StyleSheet.create({
    container: {
      flex: 1,
      height: height - 30,
      width: width - 1,
    },
    image: {
      flex: 1,
      justifyContent: 'center',
    },
    closeButton: {
      position: 'absolute',
      left: 1,
      top: 20,
    },
    previousButton: {
      position: 'absolute',
      left: 25,
      top: height * 0.5 - 55,
    },
    nextButton: {
      position: 'absolute',
      right: 25,
      top: height * 0.5 - 55,
    },
    playButton: {
      position: 'absolute',
      left: width * 0.5 - 35,
      top: height * 0.5 - 55,
    },
    controls: {
      position: 'absolute',
      bottom: 30,
      marginHorizontal: 15,
      flexDirection: 'row',
    },
    time: {
      color: 'white',
      alignSelf: 'center',
    },
    slider: {
      flex: 1,
      alignSelf: 'center',
      marginHorizontal: Platform.select({ios: 5}),
    },
    duration: {
      color: 'white',
      alignSelf: 'center',
    },
    loadingSpinner: {
      marginLeft: 15,
      marginTop: 15,
    },
  });

export default React.memo(AudioPlayer);

AudioPlayer.propTypes = {
  point: PropTypes.object.isRequired,
  disablePointControls: PropTypes.bool.isRequired,
  close: PropTypes.func.isRequired,
  showPrevious: PropTypes.func.isRequired,
  showNext: PropTypes.func.isRequired,
  isLast: PropTypes.bool.isRequired,
  isFirst: PropTypes.bool.isRequired,
};
