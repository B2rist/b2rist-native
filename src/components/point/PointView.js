import React, {useCallback} from 'react';
import {Avatar, Button, Card, Paragraph} from 'react-native-paper';
import PanelHandle from 'components/point/PanelHandle';
import PropTypes from 'prop-types';

const MapMarker = props => <Avatar.Icon {...props} icon="map-marker" />;

const PointView = ({
  point,
  distance,
  isNearby,
  setPlayingPoint,
  clearPoint,
}) => {
  const onSetPlayingPoint = useCallback(() => {
    setPlayingPoint(point);
  }, [point, setPlayingPoint]);

  return (
    <Card>
      <Card.Content>
        <PanelHandle />
      </Card.Content>
      <Card.Title
        title={point.name}
        subtitle={`${distance} m`}
        left={MapMarker}
      />
      <Card.Actions>
        {!isNearby && (
          <Button icon="google-nearby" mode="outlined" onPress={clearPoint}>
            Show nearby
          </Button>
        )}
        <Button
          icon="play-box-outline"
          mode="contained"
          onPress={onSetPlayingPoint}>
          Play
        </Button>
      </Card.Actions>
      {point.cover && <Card.Cover source={{uri: point.cover}} />}
      <Card.Content>
        <Paragraph>{point.description}</Paragraph>
      </Card.Content>
    </Card>
  );
};

export default PointView;

PointView.propTypes = {
  point: PropTypes.shape({
    name: PropTypes.string.isRequired,
    cover: PropTypes.string,
    description: PropTypes.string,
  }).isRequired,
  distance: PropTypes.number.isRequired,
  isNearby: PropTypes.bool.isRequired,
  setPlayingPoint: PropTypes.func.isRequired,
  clearPoint: PropTypes.func.isRequired,
};
