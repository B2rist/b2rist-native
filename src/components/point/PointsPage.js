import React, {
  useCallback,
  useContext,
  useEffect,
  useMemo,
  useState,
} from 'react';
import {useSelector} from 'react-redux';
import {selectPointPage} from 'services/redux/pointSlice';
import {Platform, Slider, StyleSheet, View} from 'react-native';
import GoogleMapView from 'components/googlemaps/GoogleMapView';
import {GeoLocationContext} from 'GeoLocationProvider';
import {
  Banner,
  Button,
  Card,
  IconButton,
  Avatar,
  Modal,
  Portal,
} from 'react-native-paper';
import MediaResolver from 'components/player/MediaResolver';
import {useNavigate, createSearchParams} from 'react-router-native';
import useNearbyPoint from 'components/point/useNearbyPoint';
import NewMarkers from 'components/custom/NewMarkers';

const MapMarker = props => <Avatar.Icon {...props} icon="map-marker" />;

const PlayButton = (props, onPress) => (
  <IconButton {...props} icon="play-box-outline" onPress={onPress} />
);

const PointsPage = () => {
  const navigate = useNavigate();
  const points = useSelector(state => selectPointPage(state));
  const {position, enabled: gpsEnabled} = useContext(GeoLocationContext);
  const [playingPoint, setPlayingPoint] = useState(null);
  const [played, setPlayed] = useState([]);
  const [filterVisible, setFilterVisible] = useState(false);
  const [nearbyPoint, distance, reached] = useNearbyPoint(
    points.list,
    position,
    gpsEnabled,
  );

  useEffect(() => {
    nearbyPoint &&
      reached &&
      playingPoint?.id !== nearbyPoint.id &&
      !played.includes(nearbyPoint.id) &&
      setPlayingPoint(nearbyPoint);
  }, [nearbyPoint, played, playingPoint, reached]);

  const onClose = useCallback(() => {
    setPlayed([...played, playingPoint.id]);
    setPlayingPoint(null);
  }, [played, playingPoint]);

  const onValueChanged = useCallback(
    value =>
      navigate({
        pathname: '/points',
        search: `?${createSearchParams({radius: value})}`,
      }),
    [navigate],
  );

  const onSetPlayingPoint = useCallback(point => {
    setPlayingPoint(point);
  }, []);

  const Markers = useMemo(
    () => (
      <NewMarkers points={points.list} setPlayingPoint={onSetPlayingPoint} />
    ),
    [onSetPlayingPoint, points],
  );

  return (
    <>
      <Portal>
        <Modal
          visible={!!playingPoint}
          onDismiss={onClose}
          contentContainerStyle={styles.modal}>
          {playingPoint && (
            <MediaResolver point={playingPoint} close={onClose} />
          )}
        </Modal>
      </Portal>
      <Banner
        visible={filterVisible}
        actions={[
          {
            label: 'Close',
            onPress: () => setFilterVisible(false),
          },
        ]}>
        <Slider
          onSlidingComplete={onValueChanged}
          value={+points.page.radius}
          minimumValue={0}
          maximumValue={25}
          maximumTrackTintColor="gray"
          minimumTrackTintColor={'white'}
          thumbTintColor={'white'}
          style={styles.slider}
        />
      </Banner>
      {!filterVisible && (
        <View style={styles.moreButton}>
          <Button onPress={() => setFilterVisible(true)}>... More</Button>
        </View>
      )}
      <View style={styles.container}>
        <GoogleMapView children={Markers} />
      </View>
      {nearbyPoint && (
        <Card>
          <Card.Title
            title={nearbyPoint.name}
            subtitle={`${distance} m`}
            left={MapMarker}
            right={props =>
              PlayButton(props, () => setPlayingPoint(nearbyPoint))
            }
          />
        </Card>
      )}
    </>
  );
};

const styles = StyleSheet.create({
  modal: {
    marginTop: -20,
    flex: 1,
  },
  moreButton: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
  },
  container: {
    marginTop: 10,
    height: 400,
  },
  slider: {
    width: 300,
    height: 30,
    borderRadius: 50,
    flex: 1,
    alignSelf: 'center',
    marginHorizontal: Platform.select({ios: 5}),
  },
});

export default PointsPage;
