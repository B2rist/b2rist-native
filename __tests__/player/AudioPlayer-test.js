import 'react-native';
import React from 'react';
import {points} from '../../__mock-data__/point';
import {
  render,
  screen,
  waitFor,
  fireEvent,
  act,
} from '@testing-library/react-native';
import AudioPlayer from 'components/player/AudioPlayer';
import Sound from 'react-native-sound';

const DEFAULT_PROPS = {
  point: points[2],
  close: jest.fn(),
  showPrevious: jest.fn(),
  showNext: jest.fn(),
  isLast: false,
  disablePointControls: true,
  isFirst: false,
};

describe('audio player tests', () => {
  it('audio player renders correctly', () => {
    render(<AudioPlayer {...DEFAULT_PROPS} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('loading sound test', () => {
    render(<AudioPlayer {...DEFAULT_PROPS} />);
    expect(screen.getByTestId('sound-loading')).toBeTruthy();
    expect(screen.getByTestId('close-button')).toBeTruthy();
    // check audio url
    expect(Sound.DOCUMENT).toBe(
      'https://docs.google.com/uc?export=download&id=1SPT9SwLvr89EFdxydfZ7yQ4d_jEE9yvd',
    );
  });

  it('loaded sound test', async () => {
    const {queryByTestId, getByTestId} = render(
      <AudioPlayer {...DEFAULT_PROPS} />,
    );
    await act(async () => {
      Sound.callback();
      Sound.loaded = true;
      // check hidden sound loading indicator after loading
      await waitFor(() => expect(queryByTestId('sound-loading')).toBeNull());
      await waitFor(() => expect(queryByTestId('play-button')).toBeNull());
      // check sound play function has been called
      expect(Sound.playMock).toHaveBeenCalledTimes(1);
      fireEvent.press(getByTestId('touchable-player'));
      // check play button displays on touch screen
      await waitFor(() => expect(queryByTestId('play-button')).toBeTruthy());
      expect(Sound.pauseMock).toHaveBeenCalledTimes(1);
      // play again test
      fireEvent.press(getByTestId('touchable-player'));
      await waitFor(() => expect(queryByTestId('play-button')).toBeNull());
      expect(Sound.playMock).toHaveBeenCalledTimes(2);
      fireEvent.press(screen.getByTestId('close-button'));
      expect(Sound.releaseMock).toHaveBeenCalledTimes(2);
      expect(DEFAULT_PROPS.close).toHaveBeenCalledTimes(1);
    });
  });

  it('error loading sound test', async () => {
    const {queryByTestId} = render(<AudioPlayer {...DEFAULT_PROPS} />);
    await act(async () => {
      Sound.callback('error');
      Sound.loaded = true;
      // check hidden sound loading indicator after loading
      await waitFor(() => expect(queryByTestId('sound-loading')).toBeNull());
      await waitFor(() => expect(queryByTestId('play-button')).toBeTruthy());
    });
  });

  it('image collage test', async () => {
    const {queryByTestId} = render(<AudioPlayer {...DEFAULT_PROPS} />);
    await act(async () => {
      Sound.callback();
      Sound.loaded = true;
      expect(queryByTestId('background').props.source).toStrictEqual({
        uri: 'https://drive.google.com/uc?export=view&id=1Dr1ml2-GZcA5zyf68VmiFudFAoyCu9eb',
      });
      Sound.currentTime = 20;
      await waitFor(() =>
        expect(queryByTestId('background').props.source).toStrictEqual({
          uri: 'https://drive.google.com/uc?export=view&id=1JXNtJrUOfYppjEpzIBgQ7EWTs31Bs-Dy',
        }),
      );
    });
  });

  it('point navigation button test', async () => {
    const {queryByTestId, getByTestId} = render(
      <AudioPlayer {...DEFAULT_PROPS} disablePointControls={false} />,
    );
    await act(async () => {
      Sound.callback();
      Sound.loaded = true;
      await waitFor(() => expect(queryByTestId('sound-loading')).toBeNull());
      await waitFor(() => expect(queryByTestId('play-button')).toBeNull());
      fireEvent.press(getByTestId('touchable-player'));
      await waitFor(() =>
        expect(queryByTestId('previous-button')).toBeTruthy(),
      );
      fireEvent.press(queryByTestId('previous-button'));
      expect(Sound.releaseMock).toHaveBeenCalledTimes(2);
      expect(DEFAULT_PROPS.showPrevious).toHaveBeenCalledTimes(1);
      await waitFor(() => expect(queryByTestId('next-button')).toBeTruthy());
      fireEvent.press(queryByTestId('next-button'));
      expect(Sound.releaseMock).toHaveBeenCalledTimes(2);
      expect(DEFAULT_PROPS.showNext).toHaveBeenCalledTimes(1);
    });
  });
});
