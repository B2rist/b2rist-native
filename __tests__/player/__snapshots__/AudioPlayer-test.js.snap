// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`audio player tests audio player renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "black",
      "flex": 1,
      "height": 1304,
      "opacity": 1,
      "width": 749,
    }
  }
  testID="touchable-player"
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      resizeMethod="resize"
      source={
        {
          "uri": "https://drive.google.com/uc?export=view&id=1Dr1ml2-GZcA5zyf68VmiFudFAoyCu9eb",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="background"
    />
    <View
      style={
        {
          "left": 1,
          "position": "absolute",
          "top": 20,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "alignSelf": undefined,
            "bottom": undefined,
            "end": undefined,
            "left": undefined,
            "position": undefined,
            "right": undefined,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "start": undefined,
            "top": undefined,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": undefined,
                "borderColor": "rgba(28, 27, 31, 0.12)",
                "borderRadius": 33,
                "borderWidth": 0,
                "elevation": 0,
                "height": 66,
                "margin": 6,
                "overflow": "hidden",
                "width": 66,
              }
            }
          >
            <View
              accessibilityComponentType="button"
              accessibilityRole="button"
              accessibilityState={
                {
                  "disabled": true,
                }
              }
              accessibilityTraits={
                [
                  "button",
                  "disabled",
                ]
              }
              accessible={true}
              centered={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  false,
                  [
                    {
                      "alignItems": "center",
                      "flexGrow": 1,
                      "justifyContent": "center",
                    },
                    {
                      "borderRadius": 33,
                    },
                  ],
                ]
              }
              testID="close-button"
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  [
                    {
                      "color": "rgba(28, 27, 31, 0.38)",
                      "fontSize": 50,
                    },
                    [
                      {
                        "lineHeight": 50,
                        "transform": [
                          {
                            "scaleX": 1,
                          },
                        ],
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                    ],
                    {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                ó°…–
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "left": 340,
          "position": "absolute",
          "top": 612,
        }
      }
    >
      <View
        accessibilityRole="progressbar"
        accessibilityState={
          {
            "busy": true,
          }
        }
        accessible={true}
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
            {
              "marginLeft": 15,
              "marginTop": 15,
            },
          ]
        }
        testID="sound-loading"
      >
        <View
          collapsable={false}
          style={
            {
              "height": 45,
              "opacity": 1,
              "width": 45,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "height": 45,
                  "transform": [
                    {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 45,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 22.5,
                    "overflow": "hidden",
                    "width": 45,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "height": 45,
                      "transform": [
                        {
                          "translateY": 0,
                        },
                        {
                          "rotate": "-165deg",
                        },
                      ],
                      "width": 45,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "height": 22.5,
                        "overflow": "hidden",
                        "width": 45,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "borderColor": "rgba(103, 80, 164, 1)",
                          "borderRadius": 22.5,
                          "borderWidth": 4.5,
                          "height": 45,
                          "width": 45,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "height": 45,
                  "transform": [
                    {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 45,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 22.5,
                    "overflow": "hidden",
                    "top": 22.5,
                    "width": 45,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "height": 45,
                      "transform": [
                        {
                          "translateY": -22.5,
                        },
                        {
                          "rotate": "345deg",
                        },
                      ],
                      "width": 45,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "height": 22.5,
                        "overflow": "hidden",
                        "width": 45,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "borderColor": "rgba(103, 80, 164, 1)",
                          "borderRadius": 22.5,
                          "borderWidth": 4.5,
                          "height": 45,
                          "width": 45,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "bottom": 30,
          "flexDirection": "row",
          "marginHorizontal": 15,
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            {
              "alignSelf": "center",
              "color": "white",
            },
          ]
        }
      >
        00:00:00
      </Text>
      <RNCSlider
        disabled={false}
        inverted={false}
        lowerLimit={-9007199254740991}
        maximumTrackTintColor="gray"
        maximumValue={0}
        minimumTrackTintColor="white"
        minimumValue={0}
        onChange={null}
        onRNCSliderAccessibilityAction={null}
        onRNCSliderSlidingComplete={[Function]}
        onRNCSliderSlidingStart={null}
        onRNCSliderValueChange={null}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        step={0}
        style={
          [
            {
              "alignSelf": "center",
              "flex": 1,
              "marginHorizontal": 5,
            },
            {
              "height": 40,
            },
          ]
        }
        tapToSeek={false}
        thumbTintColor="white"
        upperLimit={9007199254740991}
        value={0}
      />
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            {
              "alignSelf": "center",
              "color": "white",
            },
          ]
        }
      >
        00:00:00
      </Text>
    </View>
  </View>
</View>
`;
